### PR TITLE
PointerInput refactor

### PR DESCRIPTION
**Pull request by Hixie**
Instead of PointerInputEvent having a "type" field, we now have a
different class for each pointer type.

This has ripple effects throughout the system.

I also did code cleanup in affected files while I was there.

### CODE REVIEW

**Comment by Hixie** [comment]
Sorry it's so long. It kinda got out of control. I stopped short of doing the bindings refactor, though, so it could be worse...

File: packages/flutter/lib/src/gestures/events.dart [hunk 1]
```diff
@@ -41,58 +40,357 @@ class PointerInputEvent extends InputEvent {
...
+    bool obscured: false,
+    double pressureMin: 0.0,
+    double pressureMax: 0.0,
+    double distance: 0.0,
+    double distanceMax: 0.0,
+    double radiusMin: 0.0,
+    double radiusMax: 0.0,
+    double orientation: 0.0,
+    double tilt: 0.0
+  }) : super (
```
Comment (abarth) [inline packages/flutter/lib/src/gestures/events.dart]: extra space between `super` and `(`

File: packages/flutter/lib/src/rendering/binding.dart [hunk 2]
```diff
@@ -17,119 +17,180 @@ import 'hit_test.dart';
...
-      case PointerType.MOVE:
-        eventType = 'pointermove';
-        // state == null means the pointer is hovering
-        if (state != null) {
-          dx = position.x - state.lastPosition.x;
-          dy = position.y - state.lastPosition.y;
+  // map from platform pointer identifiers to PointerEvent pointer identifiers
+  static Map<int, _PointerState> _pointers = <int, _PointerState>{};
+
+  static Iterable<PointerEvent> expand(Iterable<Pointer> packet) sync* {
```
Comment (abarth) [inline packages/flutter/lib/src/rendering/binding.dart]: fancy

File: packages/unit/test/gestures/long_press_test.dart [hunk 3]
```diff
@@ -2,18 +2,14 @@ import 'package:quiver/testing/async.dart';
 import 'package:flutter/gestures.dart';
 import 'package:test/test.dart';
 
-final PointerInputEvent down = new PointerInputEvent(
+final PointerDownEvent down = new PointerDownEvent(
```
Comment (abarth) [inline packages/unit/test/gestures/long_press_test.dart]: const?

**Comment by abarth** [comment]
LGTM.  Would buy again.

**Comment by Hixie** [comment]
thanks!

**Comment by abarth** [comment]
Looks like you missed some in flutter_sprites.  Also, you should email flutter-dev about the API change.

**Comment by Hixie** [comment]
Woah, that means flutter_sprites is neither tested nor analyzed.

**Pull request merged**